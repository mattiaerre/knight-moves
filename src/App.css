.Knight {
  background-color: blue;
  color: white;
}

.Rank--even,
.Rank--odd {
  display: flex;
}

.Rank--even .Square:nth-child(1n) {
  background-color: lightgoldenrodyellow;
  color: olivedrab;
}

.Rank--even .Square:nth-child(2n) {
  background-color: olivedrab;
  color: white;
}

.Rank--odd .Square:nth-child(1n) {
  background-color: olivedrab;
  color: white;
}

.Rank--odd .Square:nth-child(2n) {
  background-color: lightgoldenrodyellow;
  color: olivedrab;
}

.Knight,
.Square {
  align-items: center;
  display: flex;
  justify-content: center;
  height: 80px;
  width: 80px;
}

footer {
  font-size: small;
  margin: 16px 0 0 16px;
}

ul {
  list-style: none;
  margin: 0;
  padding: 0;
}
