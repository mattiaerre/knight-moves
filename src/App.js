import classnames from 'classnames';
import './App.css';
import { files, ranks } from './domain/constants';

function where(position) {
  return position[0] === 'b' && position[1] === 1;
}

function App() {
  return (
    <div>
      {[...ranks].reverse().map((rank, index) => {
        return (
          <ul
            className={classnames({
              'Rank--even': index % 2 === 0,
              'Rank--odd': index % 2 === 1
            })}
            key={rank}
          >
            {files.map((file, index) => (
              <li
                className={classnames({
                  Knight: where([file, rank]),
                  Square: !where([file, rank])
                })}
                data-file={file}
                data-rank={rank}
                key={file}
              />
            ))}
          </ul>
        );
      })}
    </div>
  );
}

export default App;
