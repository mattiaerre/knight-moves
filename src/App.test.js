import { render } from '@testing-library/react';
import App from './App';

test('App snapshot', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
