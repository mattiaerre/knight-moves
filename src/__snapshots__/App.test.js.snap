// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<div>
  <div>
    <ul
      class="Rank--even"
    >
      <li
        class="Square"
        data-file="a"
        data-rank="8"
      >
        a
        8
      </li>
      <li
        class="Square"
        data-file="b"
        data-rank="8"
      >
        b
        8
      </li>
      <li
        class="Square"
        data-file="c"
        data-rank="8"
      >
        c
        8
      </li>
      <li
        class="Square"
        data-file="d"
        data-rank="8"
      >
        d
        8
      </li>
      <li
        class="Square"
        data-file="e"
        data-rank="8"
      >
        e
        8
      </li>
      <li
        class="Square"
        data-file="f"
        data-rank="8"
      >
        f
        8
      </li>
      <li
        class="Square"
        data-file="g"
        data-rank="8"
      >
        g
        8
      </li>
      <li
        class="Square"
        data-file="h"
        data-rank="8"
      >
        h
        8
      </li>
    </ul>
    <ul
      class="Rank--odd"
    >
      <li
        class="Square"
        data-file="a"
        data-rank="7"
      >
        a
        7
      </li>
      <li
        class="Square"
        data-file="b"
        data-rank="7"
      >
        b
        7
      </li>
      <li
        class="Square"
        data-file="c"
        data-rank="7"
      >
        c
        7
      </li>
      <li
        class="Square"
        data-file="d"
        data-rank="7"
      >
        d
        7
      </li>
      <li
        class="Square"
        data-file="e"
        data-rank="7"
      >
        e
        7
      </li>
      <li
        class="Square"
        data-file="f"
        data-rank="7"
      >
        f
        7
      </li>
      <li
        class="Square"
        data-file="g"
        data-rank="7"
      >
        g
        7
      </li>
      <li
        class="Square"
        data-file="h"
        data-rank="7"
      >
        h
        7
      </li>
    </ul>
    <ul
      class="Rank--even"
    >
      <li
        class="Square"
        data-file="a"
        data-rank="6"
      >
        a
        6
      </li>
      <li
        class="Square"
        data-file="b"
        data-rank="6"
      >
        b
        6
      </li>
      <li
        class="Square"
        data-file="c"
        data-rank="6"
      >
        c
        6
      </li>
      <li
        class="Square"
        data-file="d"
        data-rank="6"
      >
        d
        6
      </li>
      <li
        class="Square"
        data-file="e"
        data-rank="6"
      >
        e
        6
      </li>
      <li
        class="Square"
        data-file="f"
        data-rank="6"
      >
        f
        6
      </li>
      <li
        class="Square"
        data-file="g"
        data-rank="6"
      >
        g
        6
      </li>
      <li
        class="Square"
        data-file="h"
        data-rank="6"
      >
        h
        6
      </li>
    </ul>
    <ul
      class="Rank--odd"
    >
      <li
        class="Square"
        data-file="a"
        data-rank="5"
      >
        a
        5
      </li>
      <li
        class="Square"
        data-file="b"
        data-rank="5"
      >
        b
        5
      </li>
      <li
        class="Square"
        data-file="c"
        data-rank="5"
      >
        c
        5
      </li>
      <li
        class="Square"
        data-file="d"
        data-rank="5"
      >
        d
        5
      </li>
      <li
        class="Square"
        data-file="e"
        data-rank="5"
      >
        e
        5
      </li>
      <li
        class="Square"
        data-file="f"
        data-rank="5"
      >
        f
        5
      </li>
      <li
        class="Square"
        data-file="g"
        data-rank="5"
      >
        g
        5
      </li>
      <li
        class="Square"
        data-file="h"
        data-rank="5"
      >
        h
        5
      </li>
    </ul>
    <ul
      class="Rank--even"
    >
      <li
        class="Square"
        data-file="a"
        data-rank="4"
      >
        a
        4
      </li>
      <li
        class="Square"
        data-file="b"
        data-rank="4"
      >
        b
        4
      </li>
      <li
        class="Square"
        data-file="c"
        data-rank="4"
      >
        c
        4
      </li>
      <li
        class="Square"
        data-file="d"
        data-rank="4"
      >
        d
        4
      </li>
      <li
        class="Square"
        data-file="e"
        data-rank="4"
      >
        e
        4
      </li>
      <li
        class="Square"
        data-file="f"
        data-rank="4"
      >
        f
        4
      </li>
      <li
        class="Square"
        data-file="g"
        data-rank="4"
      >
        g
        4
      </li>
      <li
        class="Square"
        data-file="h"
        data-rank="4"
      >
        h
        4
      </li>
    </ul>
    <ul
      class="Rank--odd"
    >
      <li
        class="Square"
        data-file="a"
        data-rank="3"
      >
        a
        3
      </li>
      <li
        class="Square"
        data-file="b"
        data-rank="3"
      >
        b
        3
      </li>
      <li
        class="Square"
        data-file="c"
        data-rank="3"
      >
        c
        3
      </li>
      <li
        class="Square"
        data-file="d"
        data-rank="3"
      >
        d
        3
      </li>
      <li
        class="Square"
        data-file="e"
        data-rank="3"
      >
        e
        3
      </li>
      <li
        class="Square"
        data-file="f"
        data-rank="3"
      >
        f
        3
      </li>
      <li
        class="Square"
        data-file="g"
        data-rank="3"
      >
        g
        3
      </li>
      <li
        class="Square"
        data-file="h"
        data-rank="3"
      >
        h
        3
      </li>
    </ul>
    <ul
      class="Rank--even"
    >
      <li
        class="Square"
        data-file="a"
        data-rank="2"
      >
        a
        2
      </li>
      <li
        class="Square"
        data-file="b"
        data-rank="2"
      >
        b
        2
      </li>
      <li
        class="Square"
        data-file="c"
        data-rank="2"
      >
        c
        2
      </li>
      <li
        class="Square"
        data-file="d"
        data-rank="2"
      >
        d
        2
      </li>
      <li
        class="Square"
        data-file="e"
        data-rank="2"
      >
        e
        2
      </li>
      <li
        class="Square"
        data-file="f"
        data-rank="2"
      >
        f
        2
      </li>
      <li
        class="Square"
        data-file="g"
        data-rank="2"
      >
        g
        2
      </li>
      <li
        class="Square"
        data-file="h"
        data-rank="2"
      >
        h
        2
      </li>
    </ul>
    <ul
      class="Rank--odd"
    >
      <li
        class="Square"
        data-file="a"
        data-rank="1"
      >
        a
        1
      </li>
      <li
        class="Square"
        data-file="b"
        data-rank="1"
      >
        b
        1
      </li>
      <li
        class="Square"
        data-file="c"
        data-rank="1"
      >
        c
        1
      </li>
      <li
        class="Square"
        data-file="d"
        data-rank="1"
      >
        d
        1
      </li>
      <li
        class="Square"
        data-file="e"
        data-rank="1"
      >
        e
        1
      </li>
      <li
        class="Square"
        data-file="f"
        data-rank="1"
      >
        f
        1
      </li>
      <li
        class="Square"
        data-file="g"
        data-rank="1"
      >
        g
        1
      </li>
      <li
        class="Square"
        data-file="h"
        data-rank="1"
      >
        h
        1
      </li>
    </ul>
    <footer>
      knight-moves
       v
      0.5.0
    </footer>
  </div>
</div>
`;
